### PR TITLE
Security module updates

### PR DESCRIPTION
**Pull request by labkey-nicka**
#### Changes
* Adds `Security.getGroupsForCurrentUser` and `Security.getReadableContainers`.
* Updates typings to use `RequestCallbackOptions` for `request`-based APIs. 
* Add response typings and documentation for many endpoints.
* Remove reference to Sencha Connection as link is no longer valid.

### CODE REVIEW

File: src/labkey/security/Container.ts [hunk 1]
```diff
@@ -291,39 +261,45 @@ export function getHomeContainer(): string {
...
+    /** whether this module should be enabled by default for this container */
+    enabled: boolean
+
+    /** Name of the module */
+    name: string
+
+    /** Whether this module is required in the folder type specified above */
+    required: boolean
+
+    /** */
```
Comment (cnathe) [inline src/labkey/security/Container.ts:277]: to be filled in later?

File: src/labkey/security/Policy.ts [hunk 2]
```diff
@@ -63,17 +48,31 @@ export function deletePolicy(config: DeletePolicyOptions): XMLHttpRequest {
...
+        role: string
+        userId: number
+    }[]
+    modified: string
+    modifiedMillis: number
+    resourceId: string
+    requestedResourceId: string
+}
+
 export interface GetPolicyOptions {
```
Comment (cnathe) [inline src/labkey/security/Policy.ts:62]: does this one not extend RequestCallbackOptions because of the additional success parameter?

Comment (labkey-nicka) [reply src/labkey/security/Policy.ts:62]: Yes, precisely. Changing the signature of the `success` method parameters beyond the type of the initial `data` is not supported by `RequestCallbackOptions`.

File: src/labkey/security/Container.ts (hunk 1 above)
Comment (labkey-nicka) [reply src/labkey/security/Container.ts]: Oops, fixed. Really wasn't clear what it means beyond requiring site permissions...

**Pull request merged**